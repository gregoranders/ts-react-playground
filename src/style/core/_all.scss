@import '../../../node_modules/breakpoint-sass/stylesheets/_breakpoint';

@import '_colors';
@import '_variables';

@import '_w3-pro';
@import '_w3-theme';


.w3-theme-l5 {
  @include w3-theme-l5()
}

.w3-theme-l4 {
  @include w3-theme-l4()
}

.w3-theme-l3 {
  @include w3-theme-l3()
}

.w3-theme-l2 {
  @include w3-theme-l2()
}

.w3-theme-l1 {
  @include w3-theme-l1()
}

.w3-theme-d1 {
  @include w3-theme-d1()
}

.w3-theme-d2 {
  @include w3-theme-d2()
}

.w3-theme-d3 {
  @include w3-theme-d3()
}

.w3-theme-d4 {
  @include w3-theme-d4()
}

.w3-theme-d5 {
  @include w3-theme-d5()
}

.w3-theme-light {
  @include w3-theme-light()
}

.w3-theme-dark {
  @include w3-theme-dark()
}

.w3-theme-action {
  @include w3-theme-action()
}

.w3-theme {
  @include w3-theme()
}

.w3-text-theme {
  @include w3-text-theme()
}

.w3-border-theme {
  @include w3-border-theme()
}

.w3-hover-theme:hover {
  @include w3-hover-theme()
}

.w3-hover-text-theme:hover {
  @include w3-hover-text-theme()
}

.w3-hover-border-theme:hover {
  @include w3-hover-border-theme()
}
