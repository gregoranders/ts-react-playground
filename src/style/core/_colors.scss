/* https://www.w3schools.com/colors/colors_monochromatic.asp */

$color-white: #fff;
$color-black: #000;

$color-front: lighten($color-black, 10%);
$color-back: darken($color-white, 10%);

$color-blue-1: #e1ecf9;
$color-blue-2: #609ce1;
$color-blue-3: #2369b8;
$color-blue-4: #133863;
$color-blue-5: #091d34;

$color-green-1: #e8f3e8;
$color-green-2: #92c591;
$color-green-3: #549d53;
$color-green-4: #305a30;
$color-green-5: #0a3409;

$color-red-1: #fcdee0;
$color-red-2: #f37c84;
$color-red-3: #ea1f2d;
$color-red-4: #950e17;
$color-red-5: #34090c;

$color-yellow-1: #fef7dc;
$color-yellow-2: #fde281;
$color-yellow-3: #fccb1d;
$color-yellow-4: #b08a03;
$color-yellow-5: #342b09;

$color-theme-light: #f5f7f8;
$color-theme-l5:    #f5f7f8;
$color-theme-l4:    #dfe5e8;
$color-theme-l3:    #becbd2;
$color-theme-l2:    #9eb1bb;
$color-theme-l1:    #7d97a5;
$color-theme:       #607d8b;
$color-theme-d1:    #57707d;
$color-theme-d2:    #4d636f;
$color-theme-d3:    #435761;
$color-theme-d4:    #3a4b53;
$color-theme-d5:    #303e45;
$color-theme-dark:  #303e45;
