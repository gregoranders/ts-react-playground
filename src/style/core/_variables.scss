$screen-sm-min: 567px;
$screen-md-min: 768px;
$screen-lg-min: 992px;
$screen-xl-min: 1200px;
