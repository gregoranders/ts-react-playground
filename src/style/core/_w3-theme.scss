@import '_colors';

@mixin w3-theme-l5 {
  color           : $color-black !important;
  background-color: $color-theme-l5 !important
}

@mixin w3-theme-l4 {
  color           : $color-black !important;
  background-color: $color-theme-l4 !important
}

@mixin w3-theme-l3 {
  color           : $color-black !important;
  background-color: $color-theme-l3 !important
}

@mixin w3-theme-l2 {
  color           : $color-black !important;
  background-color: $color-theme-l2 !important
}

@mixin w3-theme-l1 {
  color           : $color-white !important;
  background-color: $color-theme-l1 !important
}

@mixin w3-theme-d1 {
  color           : $color-white !important;
  background-color: $color-theme-d1 !important
}

@mixin w3-theme-d2 {
  color           : $color-white !important;
  background-color: $color-theme-d2 !important
}

@mixin w3-theme-d3 {
  color           : $color-white !important;
  background-color: $color-theme-d3 !important
}

@mixin w3-theme-d4 {
  color           : $color-white !important;
  background-color: $color-theme-d4 !important
}

@mixin w3-theme-d5 {
  color           : $color-white !important;
  background-color: $color-theme-d5 !important
}

@mixin w3-theme-light {
  color           : $color-black !important;
  background-color: $color-theme-light !important
}

@mixin w3-theme-dark {
  color           : $color-white !important;
  background-color: $color-theme-dark !important
}

@mixin w3-theme-action {
  color           : $color-white !important;
  background-color: $color-theme-dark !important
}

@mixin w3-theme {
  color           : $color-white !important;
  background-color: $color-theme !important
}

@mixin w3-text-theme {
  color: $color-theme !important
}

@mixin w3-border-theme {
  border-color: $color-theme !important
}

@mixin w3-hover-theme {
  color           : $color-white !important;
  background-color: $color-theme !important
}

@mixin w3-hover-text-theme {
  color: $color-theme !important
}

@mixin w3-hover-border-theme {
  border-color: $color-theme !important
}
