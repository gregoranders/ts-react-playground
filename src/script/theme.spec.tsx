import React, { RecoilRoot, create, mount } from '@app/testUtils';

import * as TestSubject from '@app/theme';
import ThemeAtom, { Theme } from '@models/theme';
import { MuiThemeProvider } from '@material-ui/core';

describe(`${TestSubject.ThemeProvider.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    expect(testSubject.text()).toBe('ThemeProvider');
  });

  describe('themes', () => {
    it('light', () => {
      const testSubject = mount(
        <RecoilRoot
          initializeState={({ set }) => {
            set(ThemeAtom, Theme.LIGHT);
          }}
        >
          <TestComponent />
        </RecoilRoot>,
      );
      expect(testSubject.find(MuiThemeProvider)).toHaveProp('theme', TestSubject.themeLight);
    });

    it('dark', () => {
      const testSubject = mount(
        <RecoilRoot
          initializeState={({ set }) => {
            set(ThemeAtom, Theme.DARK);
          }}
        >
          <TestComponent />
        </RecoilRoot>,
      );
      expect(testSubject.find(MuiThemeProvider)).toHaveProp('theme', TestSubject.themeDark);
    });
  });
});
