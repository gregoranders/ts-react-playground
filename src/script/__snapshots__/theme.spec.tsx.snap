// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider snapshot 1`] = `
<h1>
  ThemeProvider
</h1>
`;
